diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index d948b27a..1d586423 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -235,7 +235,9 @@ ExternalProject_Add(cugunrock
   GIT_TAG           fea_full_bc      # provide a branch, a tag, or even a commit hash
   PREFIX            ${CUGUNROCK_DIR}
   CMAKE_ARGS        -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
-                    -DGPU_ARCHS=""
+                    -DCUDA_AUTODETECT_GENCODE=ON
+                    -DGENCODE=-gencode=arch=compute_72,code=sm_72
+                    -DCUDA_ARCHS=-gencode=arch=compute_62,code=sm_62
                     -DGUNROCK_BUILD_SHARED_LIBS=OFF
                     -DGUNROCK_BUILD_TESTS=OFF
   BUILD_BYPRODUCTS  ${CUGUNROCK_DIR}/lib/libgunrock.a
